src/libraries/System.Text.Json/tests/Serialization/CustomConverterTests/CustomConverterTests.Dynamic.Sample.Tests.cs

```diff
@@ -0,0 +1,333 @@
+
+            // Array
+            List<object> list = new List<object>();
+            list.Add(1);
+            list.Add(2);
+
+            dynamic dynamicArray = new JsonDynamicArray(list, options);
+            json = JsonSerializer.Serialize(dynamicArray, options);
+            Assert.Equal("[1,2]", json);
+
+            // Object
+            Dictionary<string, object> dictionary = new Dictionary<string, object>();
+            dictionary["One"] = 1;
+            dictionary["Two"] = 2;
+
+            dynamic dynamicObject = new JsonDynamicObject(dictionary, options);
+            json = JsonSerializer.Serialize(dynamicObject, options);
+            JsonTestHelper.AssertJsonEqual("{\"One\":1,\"Two\":2}", json);
+        }
+
+        [Fact]
+        public static void DirectUseOfDynamicTypes_Deserialize()
+        {
+            var options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+
+            JsonSerializer.Deserialize<JsonDynamicType>("{}", options);
+            JsonSerializer.Deserialize<JsonDynamicArray>("[]", options);
+            JsonSerializer.Deserialize<JsonDynamicBoolean>("true", options);
+            JsonSerializer.Deserialize<JsonDynamicNumber>("0", options);
+            JsonSerializer.Deserialize<JsonDynamicObject>("{}", options);
+            JsonSerializer.Deserialize<JsonDynamicString>("\"str\"", options);
+        }
+
+        [Fact]
+        public static void DirectUseOfDynamicTypes_AsObject_Deserialize()
+        {
+            var options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+
+            Assert.IsType<JsonDynamicArray>(JsonSerializer.Deserialize<object>("[]", options));
+            Assert.IsType<JsonDynamicBoolean>(JsonSerializer.Deserialize<object>("true", options));
+            Assert.IsType<JsonDynamicNumber>(JsonSerializer.Deserialize<object>("0", options));
+            Assert.IsType<JsonDynamicObject>(JsonSerializer.Deserialize<object>("{}", options));
+            Assert.IsType<JsonDynamicString>(JsonSerializer.Deserialize<object>("\"str\"", options));
+        }
+
+        [Fact]
+        public static void VerifyMutableDom()
+        {
+            const string ExpectedJson ="{\"MyString\":\"Hello!\",\"MyNull\":null,\"MyBoolean\":false,\"MyArray\":[\"2\"]," +
+                "\"MyInt\":43,\"MyDateTime\":\"2020-07-08T00:00:00\",\"MyGuid\":\"ed957609-cdfe-412f-88c1-02daca1b4f51\"," +
+                "\"MyObject\":{\"MyString\":\"Hello!!\"},\"Child\":{\"ChildProp\":1}}";
+
+            var options = new JsonSerializerOptions();
+            options.EnableDynamicTypes();
+
+            dynamic obj = JsonSerializer.Deserialize<dynamic>(DynamicTests.Json, options);
+            Assert.IsType<JsonDynamicObject>(obj);
+
+            // Change some values
+            obj.MyString = "Hello!";
+            obj.MyBoolean = false;
+            obj.MyInt = 43;
+            obj.MyObject = new ExpandoObject();
```

Comment:
Do people need to "know" to use ExpandoObject here?  Is there anything we can do to make this more intuitive?

Reply:
ExpandoObject is not required. I've updated the sample to use `JsonDynamicObject` instead and [added some comments around that](https://github.com/dotnet/runtime/pull/42097/files#diff-08965ed01d8e79b6bc7124ef919e91a542f3b8ac58e011b6a83f49fd39cc9776R199-R200).

The sample was also updated to make it easier to use with the non-dynamic cases where the `JsonType`-derives classes can be used directly without the overhead of `dynamic`. These non-dynamic cases are not the primary goal of this sample (since it was for `dynamic`), but are useful nevertheless for a writeable DOM without being forced to use `dynamic`.